### PR TITLE
Comparison tests for PYACTION

### PR DESCRIPTION
**Pull request by lisajulia**
Add tests that compare flow runs with the PYACTION keyword with the corresponding runs with the ACTIONX keyword.
Depends on https://github.com/OPM/opm-common/pull/3999

### CODE REVIEW

File: tests/run-comparison.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    e) EXE_NAME=${OPTARG} ;;
+    y) IGNORE_EXTRA_KW=${OPTARG} ;;
+  esac
+done
+shift $(($OPTIND-1))
+TEST_ARGS="$@"
+
+mkdir -p ${RESULT_PATH}
+cd ${RESULT_PATH}
+${BINPATH}/${EXE_NAME} ${INPUT_DATA_PATH1}/${FILENAME1} ${TEST_ARGS} --output-dir=${RESULT_PATH}
```
Comment (akva2) [inline tests/run-comparison.sh:47]: should exit early if first simulator returns non-zero status.

File: compareECLFiles.cmake [hunk 2]
```diff
@@ -95,6 +95,38 @@ function(add_test_compareECLFiles)
                         TESTNAME ${PARAM_CASENAME})
 endfunction()
 
+function(add_test_compareNewECLFiles)
```
Comment (akva2) [inline compareECLFiles.cmake:98]: i think a better name should be found, what does 'new' mean?

File: pyactionComparisons.cmake [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+add_test_compareNewECLFiles(CASENAME pyaction_wfac_insert_kw
+                         DIR1 pyaction
+                         FILENAME1 PYACTION_WFAC_INSERT_KW
+                         DIR2 actionx
+                         FILENAME2 ACTIONX_WFAC
+                         SIMULATOR flow
+                         ABS_TOL ${abs_tol}
+                         REL_TOL ${rel_tol}
+                         IGNORE_EXTRA_KW BOTH)
```
Comment (akva2) [inline pyactionComparisons.cmake:66]: hmm, missing newline at eol?

**Review by akva2 — COMMENTED** [review]
run-comparison.sh is not a very descriptive name, run-pyaction-comparison.sh maybe?

File: compareECLFiles.cmake (hunk 2 above)
Comment (lisajulia) [reply compareECLFiles.cmake:98]: Here, new means that this is not a regression test, i.e. this test compares two "new" runs and not an old result file with a new run.

**Comment by lisajulia** [comment]
> run-comparison.sh is not a very descriptive name, run-pyaction-comparison.sh maybe?

I chose run-comparison, because this function can be used to compare two arbitrary runs.
I later use it to compare a run with a PYACTION keyword with a run with an ACTIONX keyword, but run-comparison as such is not related to PYACTION.

File: compareECLFiles.cmake (hunk 2 above)
Comment (akva2) [reply compareECLFiles.cmake:98]: So maybe 'compareSeparate'.

Comment (lisajulia) [reply compareECLFiles.cmake:98]: Ok :+1:

File: tests/run-comparison.sh (hunk 1 above)
Comment (lisajulia) [reply tests/run-comparison.sh:47]: I added `set -e` in the file

File: compareECLFiles.cmake [hunk 4]
```diff
@@ -95,6 +95,38 @@ function(add_test_compareECLFiles)
                         TESTNAME ${PARAM_CASENAME})
 endfunction()
 
+function(add_test_compareSeparateECLFiles)
+  set(oneValueArgs CASENAME FILENAME1 FILENAME2 DIR1 DIR2 SIMULATOR ABS_TOL REL_TOL IGNORE_EXTRA_KW DIR_PREFIX)
+  set(multiValueArgs TEST_ARGS)
+  cmake_parse_arguments(PARAM "$" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
+  if(NOT PARAM_PREFIX)
+    set(PARAM_PREFIX compareNewECLFiles)
```
Comment (akva2) [inline compareECLFiles.cmake:103]: hmm

Comment (lisajulia) [reply compareECLFiles.cmake]: Thanks!

**Review by akva2 — APPROVED** [review]

**Comment by lisajulia** [comment]
jenkins build this please

**Comment by lisajulia** [comment]
@blattms Can you have another look at this PR? Thanks!

**Comment by blattms** [comment]
Approved by @akva2. Merging

**Pull request merged**